src/bloqade/analysis/address/impls.py

```diff
@@ -73,8 +75,15 @@ def new_ilist(
 class PyIndexing(interp.MethodTable):
     @interp.impl(py.GetItem)
     def getitem(self, interp: AddressAnalysis, frame: interp.Frame, stmt: py.GetItem):
-        # Integer index into the thing being indexed
-        idx = interp.get_const_value(int, stmt.index)
+        stmt_index = stmt.index
+        if stmt_index.type.is_subseteq(ilist.IListType):
```

Comment:
Specifically, I'm not sure about this part here.

Reply:
Wait when is an argument to `obj[...]` an ilist?

Reply:
Wait but this has to be a bug then since `q[0]` the should lower to a getitem with the register `q`  being `obj` not `index`.

Reply:
Oh, and the code here is completely wrong. I'm now fetching the `BlockArgument` index, which is just the index in the block. That doesn't make sense.

How do you unwrap a `BlockArgument` to get the underlying value?

Reply:
It should have a value in the analysis frame.

Reply:
I think in this case the value for the block argument is`NoQubit`. The Analysis will fail for this example at the moment.

Reply:
@weinbe58 and that is fine? If so, that also means I can't make the fidelity analysis work for nested kernels such as this one, since we're tracking the atom survivability per qubit. Maybe we should just remove that for now? Then I wouldn't have to rely on the address analysis at all.